src/Time/Slab.hpp

```diff
@@ -0,0 +1,137 @@
+
+  /// Construct a slab running between two times (exactly).
+  Slab(double start, double end) noexcept : start_(start), end_(end) {
+    ASSERT(start_ < end_, "Backwards Slab");
+  }
+
+  /// Construct a slab with a given start time and duration.  The
+  /// actual duration may differ by roundoff from the supplied value.
+  static Slab new_with_duration(double start, double duration) noexcept {
+    return Slab(start, start + duration);
+  }
+
+  /// Construct a slab with a given end time and duration.  The
+  /// actual duration may differ by roundoff from the supplied value.
+  static Slab new_with_duration_reverse(double end, double duration) noexcept {
+    return Slab(end - duration, end);
+  }
+
+  Time start() const noexcept;
+  Time end() const noexcept;
+  TimeDelta duration() const noexcept;
+
+  /// Create a new slab immediately following this one with the same
+  /// (up to roundoff) duration.
+  Slab advance() const noexcept { return Slab(end_, end_ + (end_ - start_)); }
+
+  /// Create a new slab immediately preceeding this one with the same
+  /// (up to roundoff) duration.
+  Slab retreat() const noexcept {
+    return Slab(start_ - (end_ - start_), start_);
+  }
+
+  /// Create a slab adjacent to this one in the direction indicated by
+  /// the argument, as with advance() or retreat().
+  Slab advance_towards(const TimeDelta& dt) const noexcept;
+
+  /// Create a new slab with the same start time as this one with the
+  /// given duration (up to roundoff).
+  Slab with_duration(double duration) const noexcept {
+    return Slab(start_, start_ + duration);
+  }
+
+  /// Create a new slab with the same end time as this one with the
+  /// given duration (up to roundoff).
+  Slab with_duration_reverse(double duration) const noexcept {
+    return Slab(end_ - duration, end_);
+  }
+
+  /// Check if this slab is immediately followed by the other slab.
+  bool is_followed_by(const Slab& other) const noexcept {
+    return end_ == other.start_;
+  }
+
+  /// Check if this slab is immediately preceeded by the other slab.
+  bool is_preceeded_by(const Slab& other) const noexcept {
+    return other.is_followed_by(*this);
+  }
+
+  void pup(PUP::er& p) noexcept;
+
+ private:
+  double start_;
+  double end_;
+
+  friend class Time;
+  friend class TimeDelta;
+
+  friend bool operator==(const Slab&, const Slab&) noexcept;
+  friend bool operator<(const Slab&, const Slab&) noexcept;
+  friend bool operator==(const Time&, const Time&) noexcept;
+};
+
+inline bool operator==(const Slab& a, const Slab& b) noexcept {
+  return a.start_ == b.start_ and a.end_ == b.end_;
+}
+inline bool operator!=(const Slab& a, const Slab& b) noexcept {
+  return not(a == b);
+}
+
+/// Slab comparison operators give the time ordering.  Overlapping
+/// slabs should not be compared (and will trigger an assertion).
+//@{
+inline bool operator<(const Slab& a, const Slab& b) noexcept {
+  ASSERT(a == b or a.end_ <= b.start_ or a.start_ >= b.end_,
```

Comment:
I'll update the comment, and I'll make sure to include the reasoning behind this in the overall Slab/Time explanation for the doxygen group.